src/main/java/org/checkerframework/checker/dependencyinjection/DependencyInjectionTransfer.java

```diff
@@ -1,11 +1,132 @@
+import org.checkerframework.dataflow.expression.JavaExpression;
 import org.checkerframework.framework.flow.CFAnalysis;
-import org.checkerframework.framework.flow.CFTransfer;
+import org.checkerframework.framework.flow.CFStore;
+import org.checkerframework.framework.flow.CFValue;
+import org.checkerframework.framework.type.AnnotatedTypeMirror;
+import org.checkerframework.javacutil.AnnotationMirrorSet;
+import org.checkerframework.javacutil.AnnotationUtils;
 
-public class DependencyInjectionTransfer extends CFTransfer {
+public class DependencyInjectionTransfer extends AccumulationTransfer {
+
+  private final DependencyInjectionAnnotatedTypeFactory diATF;
 
   public DependencyInjectionTransfer(CFAnalysis analysis) {
     super(analysis);
+    this.diATF = (DependencyInjectionAnnotatedTypeFactory) analysis.getTypeFactory();
+  }
+
+  @Override
+  public TransferResult<CFValue, CFStore> visitMethodInvocation(
+      final MethodInvocationNode node, final TransferInput<CFValue, CFStore> input) {
+
+    TransferResult<CFValue, CFStore> result = super.visitMethodInvocation(node, input);
+
+    if (diATF.isBindMethod(node.getTree())) {
+      Node boundClass = node.getArgument(0);
+
+      AnnotatedTypeMirror boundClassTypeMirror =
+          this.diATF
+              .getTypeFactoryOfSubchecker(ClassValChecker.class)
+              .getAnnotatedType(boundClass.getTree());
+
+      List<String> classNames =
+          AnnotationUtils.getElementValueArray(
+              boundClassTypeMirror.getAnnotation(), diATF.classValValueElement, String.class);
+
+      accumulate(node, result, classNames.toArray(new String[1]));
+
+    } else if (diATF.isAnnotatedWithMethod(node.getTree())) {
+
+      MethodInvocationNode methodInvocationNode = node;
+
+      CFValue value =
+          diATF
+              .getStoreBefore(node)
+              .getValue(JavaExpression.fromNode(methodInvocationNode.getTarget().getReceiver()));
+
+      if (value != null && !value.getAnnotations().isEmpty()) {
+        AnnotationMirrorSet annotations = value.getAnnotations();
+        annotations.forEach(
+            (annotation) -> {
+              if (AnnotationUtils.areSameByName(annotation, Bind.NAME)) {
+                List<String> classNames =
+                    AnnotationUtils.getElementValueArray(
+                        annotation, diATF.bindValValueElement, String.class);
+                MethodInvocationNode namesClassNode =
+                    (MethodInvocationNode) methodInvocationNode.getArgument(0);
+                StringLiteralNode givenName = (StringLiteralNode) namesClassNode.getArgument(0);
+
+                accumulateBindAnnotatedWith(node, result, classNames.get(0), givenName.getValue());
+              }
+            });
+      }
+    }
+
+    return result;
+  }
+
+  @Override
+  public TransferResult<CFValue, CFStore> visitAssignment(
+      AssignmentNode node, TransferInput<CFValue, CFStore> input) {
+
```

Comment:
Include a TODO here describing our attempts to investigate this and why we gave up.